### PR TITLE
Dop-969: Add an environment variable to support emitting diagnostics in JSON

### PR DESCRIPTION
**Pull request by madelinezec**
https://jira.mongodb.org/browse/DOP-969

### CODE REVIEW

File: snooty/diagnostics.py [hunk 1]
```diff
@@ -46,6 +48,14 @@ def severity(self) -> "Diagnostic.Level":
     def severity_string(self) -> str:
         return self.severity.name.title()
 
+    def serialize(self) -> n.SerializedNode:
+        diag: Dict[str, SerializableType] = {}
+
+        diag["severity"] = self.severity_string.upper()
+        diag["start"] = str(self.start[0])
```
Comment (i80and) [inline snooty/diagnostics.py:55]: What's the rationale behind converting this to a string?

File: snooty/main.py [hunk 2]
```diff
@@ -81,16 +82,22 @@ def on_progress(self, progress: int, total: int, message: str) -> None:
         pass
 
     def on_diagnostics(self, path: FileId, diagnostics: List[Diagnostic]) -> None:
+        output = os.environ.get("DiagnosticsOutput", None)
```
[the remark this answers is not in the document]
Comment (madelinezec) [reply snooty/main.py:85]: Yeah I struggled with a name. `DIAGNOSTICS_OUTPUT_JSON` ?

[the remark this answers is not in the document]
Comment (madelinezec) [reply snooty/main.py:85]: or `DIAGNOSTICS_ENV`?

File: snooty/diagnostics.py [hunk 3]
```diff
@@ -46,6 +48,14 @@ def severity(self) -> "Diagnostic.Level":
     def severity_string(self) -> str:
         return self.severity.name.title()
 
+    def serialize(self) -> n.SerializedNode:
+        diag: Dict[str, SerializableType] = {}
+        """Create dict containing diagnostic attributes for neatly reporting diagnostics at program completion"""
```
Comment (i80and) [inline snooty/diagnostics.py:53]: Docstrings go right after the function header; that's what tells Python to attach it to the function object

File: snooty/main.py [hunk 4]
```diff
@@ -9,13 +9,19 @@
   -h --help                 Show this screen.
   --commit=<commit_hash>    Commit hash of build.
   --patch=<patch_id>        Patch ID of build. Must be specified with a commit hash.
+
+Environment variables:
+  SNOOTY_PARANOID           0, 1 where 0 is default
+  DIAGNOSTICS_FORMAT        JSON, None where None is default
```
Comment (i80and) [inline snooty/main.py:15]: `None` might not be the best description of this. Perhaps `text`?

File: snooty/test_main.py [hunk 5]
```diff
@@ -14,20 +16,59 @@ def test_print(*values: Any, **kwargs: Any) -> None:
...
+        UnknownSubstitution("unknown substitution warning", 10),
+    ]
+    try:
+        backend.on_diagnostics(FileId("foo/bar.rst"), test_diagnostics[0:2])
+        backend.on_diagnostics(FileId("foo/foo.rst"), test_diagnostics[2:])
+        assert backend.total_warnings == 3
+    finally:
+        builtins.print = orig_print
+
+    # assert messages == [
```
Comment (i80and) [inline snooty/test_main.py:31]: Mm? Why are these commented out?

**Pull request merged**